integration/tests/caching/credential_caching_test.go

```diff
@@ -0,0 +1,102 @@
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+
+	. "github.com/weaveworks/eksctl/integration/runner"
+	"github.com/weaveworks/eksctl/integration/tests"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/testutils"
+)
+
+var params *tests.Params
+
+func init() {
+	// Call testing.Init() prior to tests.NewParams(), as otherwise -test.* will not be recognised. See also: https://golang.org/doc/go1.13#testing
+	testing.Init()
+	params = tests.NewParams("cache")
+}
+
+func TestCredentialsCaching(t *testing.T) {
+	testutils.RegisterAndRun(t)
+}
+
+var _ = Describe("", func() {
+	var (
+		clusterName string
+	)
+	BeforeSuite(func() {
+		clusterName = params.NewClusterName("cache")
+		cmd := params.EksctlCreateCmd.WithArgs(
+			"cluster",
+			"--name", clusterName,
+			"--without-nodegroup",
+		).WithoutArg("--region", params.Region)
+		Expect(cmd).Should(RunSuccessfully())
+	})
+	AfterSuite(func() {
+		params.DeleteClusters()
+		gexec.KillAndWait()
+	})
+	Context("accessing cluster related information", func() {
+		When("credential caching is disabled", func() {
+			var tmp string
+			BeforeEach(func() {
+				tmp, err := ioutil.TempDir("", "caching_creds")
+				Expect(err).NotTo(HaveOccurred())
+				_ = os.Setenv(eks.EksctlCacheFilenameEnvName, filepath.Join(tmp, "credentials.yaml"))
+				// Make sure the environment property is not set on the running environment.
+				_ = os.Setenv(eks.EksctlGlobalEnableCachingEnvName, "")
+			})
+			AfterEach(func() {
+				_ = os.RemoveAll(tmp)
+			})
+			It("should not cache the credentials", func() {
+				cmd := params.EksctlGetCmd.WithArgs(
+					"cluster",
+					"--name", clusterName,
+				).WithoutArg("--region", params.Region)
+				Expect(cmd).Should(RunSuccessfully())
+
+				_, err := os.Stat(filepath.Join(tmp, "credentials.yaml"))
+				Expect(os.IsNotExist(err)).To(BeTrue())
+			})
+		})
+		// Note: This proves to be a challenge since normal providers like, static and file providers, do not support
+		// expiry, therefore, they cannot be cached. This requires an AssumeRole or an EC2 role provider.
+		// For now, this part is tested via unit tests.
+		XWhen("credential caching is enabled", func() {
+			var tmp string
+			BeforeEach(func() {
+				tmp, err := ioutil.TempDir("", "caching_creds")
+				Expect(err).NotTo(HaveOccurred())
+				_ = os.Setenv(eks.EksctlCacheFilenameEnvName, filepath.Join(tmp, "credentials.yaml"))
+				_ = os.Setenv(eks.EksctlGlobalEnableCachingEnvName, "1")
+			})
+			AfterEach(func() {
+				_ = os.RemoveAll(tmp)
+			})
+			It("should cache the credentials", func() {
+				cmd := params.EksctlGetCmd.WithArgs(
+					"cluster",
+					"--name", clusterName,
+				).WithoutArg("--region", params.Region)
+				Expect(cmd).Should(RunSuccessfully())
+
+				content, err := ioutil.ReadFile(filepath.Join(tmp, "credentials.yaml"))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(content).NotTo(BeEmpty())
+			})
```

Comment:
can we test if the caching actually works, instead just if the file exists or not? or does the `get` command existing successfully mean the caching is working (it runs successfully in both scenarios so I'm guessing no)

Reply:
This test isn't working and needs another ticket to investigate how it could be made working. I need an AssumeRole provider and a user with creds that can be cached. I'm not sure how to achieve that from an integration test. Feel free to share if you have any ideas?

Reply:
no idea :/

Reply:
![YourMyOnlyHope-vblog](https://user-images.githubusercontent.com/182850/133223316-4b5434a8-4d46-4f6b-920b-68fd4d3c0f52.png)